source/draw/gpu/opengl/OpenGLFence.ooc

```diff
@@ -46,4 +47,19 @@ OpenGLFence: class extends GpuFence {
 		this _mutex unlock()
 	}
 }
+
+OpenGLPromise: class extends Promise {
+	_fence: OpenGLFence
+	init: func (=_fence) { super() }
+	free: override func {
+		this _fence free()
+		super()
+	}
+	sync: func { this _fence sync() }
+	wait: override func -> Bool { this _fence wait() }
+	wait: override func ~timeout (seconds: Double) -> Bool {
```

Comment:
Maybe functions like `wait` could use a `TimeSpan` instead of double ? I think it could be more readable, without these 1_000_000_000 in the code